src/vm/comutilnative.cpp

```diff
@@ -1922,6 +1922,21 @@ FCIMPL0(int, GCInterface::GetMaxGeneration)
 }
 FCIMPLEND
 
+FCIMPL0(INT64, GCInterface::GetAllocatedBytesForCurrentThread)
+{
+    FCALL_CONTRACT;
+
+    INT64 currentAllocated = 0;
+    Thread *pThread = GetThread();
+    if (pThread)
+    {
+        alloc_context* ac = pThread->GetAllocContext();
+        currentAllocated = ac->alloc_bytes + ac->alloc_bytes_loh - (ac->alloc_limit - ac->alloc_ptr);
```

Comment:
Is this the desired behavior? It means that this API won't work if folks deploy to small VM that has just single core assigned to it.

Would it make sense to unconditionally enable allocation contexts for CoreCLR as part of this - we have them unconditionally enabled for CoreRT already.